core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -506,17 +505,16 @@ private[spark] class DAGScheduler(
       rdd: RDD[_]): HashSet[ShuffleDependency[_, _, _]] = {
     val parents = new HashSet[ShuffleDependency[_, _, _]]
     val visited = new HashSet[RDD[_]]
-    val waitingForVisit = new ArrayStack[RDD[_]]
-    waitingForVisit.push(rdd)
+    val waitingForVisit = ListBuffer[RDD[_]](rdd)
```

Comment:
OK I think this 'optimization' does actually cause a problem. It's not happy about the two unbound wildcard types involved in one expression now. Breaking it back out ...